back-end/controllers/UserController.js

```diff
@@ -1,25 +1,72 @@
-// const User = require('../models/User');
+const jwt = require('jsonwebtoken');
+const User = require('../models/User');
+const globalUtil = require('../options.global');
 
 module.exports = {
 	async login(req, res) {
-		// const { email, password } = req.body;
+		try {
+			// Check if user exists
+			const userFound = await User.findAccount(req.body);
 
-		// Check if user exists
+			if (userFound) {
+				const sessionToken = jwt.sign(
+					{ email: req.body.email },
+					globalUtil.jwt_secret_key,
+					{ expiresIn: globalUtil.session_expired });
+				return res.status(200).json({
+					sessionToken,
+					loginStatus: true,
+					message: 'User login successfully',
+				});
+			}
+			return res.status(401).json({ loginStatus: false, message: 'User not found' });
+		} catch (error) {
+			// Handle any errors that occur during the login process
+			console.error('Error occurred during login:', error);
+			return res.status(401).json({ message: `Failed to login the user ${error}` });
+		}
+	},
 
-		// Verify password
+	async register(req, res) {
+		try {
+			const userFound = await User.findAccount(req.body);
+			if (!userFound) {
+				const registerStatus = await User.registerUser(req.body);
 
-		// Generate JWT
+				// if (registerStatus.emailAlreadyExists) {
+				// 	return res.status(409).json({ message: 'Email already exists' });
+				// }
 
-		// Handle error
+				// TODO: Check strong password
 
-		res.status(405).json({ message: 'Login is not supported via POST requests' });
-	},
-
-	async register(req, res) {
-		res.status(405).json({ message: 'Login is not supported via POST requests' });
+				if (registerStatus) {
+					const sessionToken = jwt.sign(
+						{ email: req.body.email },
+						globalUtil.jwt_secret_key,
+						{ expiresIn: globalUtil.session_expired });
+					return res.status(200).json({
+						sessionToken,
+						registerStatus: true,
+						message: 'Registered done'
+					});
+				}
+				return res.status(401).json({ message: 'Registered failed' });
+			}
+			return res.status(401).json({ message: 'Account already exists' });
+		} catch (error) {
+			console.error('Failed to register:', error);
+			return res.status(405).json({ message: `Failed to register new user. ${error}` });
```

Comment:
According to Chat GPT , we should return status 400 when Failed to register new user haha

Reply:
Good catch 😆 
By the defintion:
- The 405 error happens when a client attempts to use an HTTP method that is not allowed `by the server`
 - The 400 error happens when the server cannot or will not process the request due to something that is perceived to be a `client error`

So this is partly my fault for not taking care of all the errors when registering the user. My initial thought is to let `UserController#register` to validate variables that user input, and the `models/User#registerUser` will handle server error. 

With that being said, the `try/catch` in the `UserController#register` will actually throw error from the server and it should be 405 (a.k.a register new record in the database). 

We can log the ticket for this to go back into it once we develop a throughout system. What do you think ?